src/lib/titled-hoc.jsx

```diff
@@ -12,26 +26,89 @@ const TitledHOC = function (WrappedComponent) {
             bindAll(this, [
                 'handleUpdateProjectTitle'
             ]);
-            this.state = {
-                projectTitle: null
-            };
+        }
+        componentDidMount () {
+            this.setReduxTitle(this.props.projectTitle);
+        }
+        componentDidUpdate (prevProps) {
+            if (this.props.projectTitle !== prevProps.projectTitle) {
+                this.setReduxTitle(this.props.projectTitle);
+            }
+            if (this.props.isShowingWithoutId && !prevProps.isShowingWithoutId) {
+                const defaultProjectTitle = this.titleWithDefault();
+                this.setReduxTitle(defaultProjectTitle);
+                this.props.onUpdateProjectTitle(defaultProjectTitle);
+            }
+        }
+        titleWithDefault (title) {
```

Comment:
Why is this a function that takes a `title` argument? It looks like it's only called in one place, without an argument.

Reply:
I think I meant to refactor setReduxTitle further... doing this now

Reply:
Done